okhttp/src/main/java/okhttp3/HttpUrl.kt

```diff
@@ -1144,15 +1162,17 @@ class HttpUrl internal constructor(builder: Builder) {
       // Scheme.
       val schemeDelimiterOffset = schemeDelimiterOffset(input, pos, limit)
       if (schemeDelimiterOffset != -1) {
-        if (input.regionMatches(pos, "https:", 0, 6, ignoreCase = true)) {
-          this.scheme = "https"
-          pos += "https:".length
-        } else if (input.regionMatches(pos, "http:", 0, 5, ignoreCase = true)) {
-          this.scheme = "http"
-          pos += "http:".length
-        } else {
-          throw IllegalArgumentException("Expected URL scheme 'http' or 'https' but was '"
-              + input.substring(0, schemeDelimiterOffset) + "'")
+        when {
```

Comment:
Not sure about this one. How about
```kotlin
scheme = when {
  ...
}
pos += scheme.length
```

Reply:
scheme is changeable in between, so ended up with an awkward also, or a !! check

Reply:
Maybe read to local?
```kotlin
val scheme = when {
  // ...
}
this.scheme = scheme
pos += scheme.length
```